src/components/TextInput/TextInput.stories.tsx

```diff
@@ -0,0 +1,284 @@
+    showRequiredLabel: { table: { defaultValue: { summary: true } } },
+    step: { table: { defaultValue: { summary: 1 } } },
+    textInputType: {
+      control: false,
+      table: { defaultValue: { summary: "default" } },
+    },
+    type: {
+      control: { type: "select" },
+      options: textInputTypesArray,
+      table: { defaultValue: { summary: "text" } },
+    },
+    value: { control: false },
+  },
+};
+
+export default meta;
+type Story = StoryObj<typeof TextInput>;
+
+/**
+ * Main Story for the TextInput component. This must contains the `args`
+ * and `parameters` properties in this object.
+ */
+export const WithControls: Story = {
+  args: {
+    className: undefined,
+    defaultValue: undefined,
+    helperText: "Choose wisely.",
+    id: "textInput-id",
+    isClearable: false,
+    isClearableCallback: undefined,
+    isDisabled: false,
+    isInvalid: false,
+    invalidText: "This is error text :(",
+    isRequired: true,
+    labelText: "What is your favorite color?",
+    max: undefined,
+    maxLength: undefined,
+    min: undefined,
+    name: undefined,
+    onChange: undefined,
+    onClick: undefined,
+    onFocus: undefined,
+    pattern: undefined,
+    placeholder: "e.g. blue, green, etc.",
+    requiredLabelText: undefined,
+    showHelperInvalidText: true,
+    showLabel: true,
+    showRequiredLabel: true,
+    step: 1,
+    textInputType: "default",
+    type: "text",
+    value: undefined,
+  },
+  render: (args) => <TextInput {...args} />,
+  parameters: {
+    design: {
+      type: "figma",
+      url: "https://www.figma.com/file/qShodlfNCJHb8n03IFyApM/Master?node-id=11895%3A547",
+    },
+    jest: "TextInput.test.tsx",
+  },
+};
+
+// The following are additional TextInput example Stories.
+export const LabellingVariations: Story = {
+  render: () => (
+    <VStack align="stretch" spacing={12}>
+      <TextInput
+        helperText="Choose wisely!"
+        id="textInput-1"
+        labelText="What is your favorite color?"
+        placeholder="i.e. blue, green, etc."
+      />
+      <TextInput
+        id="textInput-2"
+        labelText="What is your favorite color?"
+        placeholder="i.e. blue, green, etc."
+        showLabel={false}
+      />
+      <TextInput
+        id="textInput-3"
+        isRequired
+        labelText="What is your favorite color?"
+        placeholder="i.e. blue, green, etc."
+      />
+      <TextInput
+        id="textInput-4"
+        isRequired
+        labelText="What is your favorite color?"
+        placeholder="i.e. blue, green, etc."
+        showRequiredLabel={false}
+      />
+      <TextInput
+        helperText="Choose wisely!"
+        id="textInput-5"
+        labelText="What is your favorite color?"
+        placeholder="i.e. blue, green, etc."
+        showLabel={false}
+      />
+    </VStack>
+  ),
+};
+
+export const BrowserStates: Story = {
+  render: () => (
+    <VStack align="stretch" spacing="s">
```

Comment:
Similar to the formatting we used on other stories, wrap each heading/component combo in a `div` or `Box` and then bump the `VStack` spacing up to `l`.